Site changelogs.

sprunk opened the pull request:
WIP
* there's a bunch of FIXMEs that I couldnt figure out
* it would be good not to merge until #2257 is done so as not to pile more work there

saurtron commented on doc/site/changelogs/changelog-2025-03.markdown:
```diff
@@ -18,15 +18,18 @@ Existing shaders should generally keep working.
 * archive cache was reworked. Should be much faster to process, but will take up more disk space and it's not yet known how stable it is.
 * added GL debug annotations to main engine GL draw calls, this is similar to tracy scopes for regular code.
 The caveat here is that this can have a large performance cost in `/debugGL` mode. The next release is planned to feature improvements in this area.
+* the default value for `FontOutlineWidth` springsetting changed from 3 to 2. This restores how it looked a few versions earlier.
```
just makes it look like in last release, it was changed in between versions because of fixed blur algo by @lhog, that's all.

lhog commented on doc/site/changelogs/changelog-2025-03.markdown:
```diff
@@ -171,6 +183,8 @@ The practical effect is that the Engine table is available in some LuaParser env
 * add `experience.experienceGrade` number modrule, same as calling Spring.SetExperienceGrade.
 * the `allowHoverUnitStrafing` modrule now defaults to `false`. Previously it defaulted to `false` for HAPFS and `true` for QTPFS.
 * bumpmapped water (aka `/water 4`) now has a different default texture.
+* add a sync debug mode that dumps much more data on a desync. Requires a separate exe compiled with the option.
```
`SYNC_HISTORY` is currently forced on. The history will be dumped to the same desync files as before named:
`ClientGameState-<random number>-[<desync frame>-<desync frame>].txt`

saurtron commented on doc/site/changelogs/changelog-2025-03.markdown:
```diff
@@ -18,15 +18,18 @@ Existing shaders should generally keep working.
 * units no longer drop Guard commands if the target was out of map for more than 5s. For a replacement look for `control_guard.lua` in basecontent.
 * fixed the `SPRING_LOG_SECTIONS` environment var, it no longer requires a comma in front.
 * `gl.GetAtmosphere("skyDir")` now returns nil. The skyDir was never actually used for anything.
+* FIXME some info shader changes at commit be96560770b95cfe18a6c5ff3cbd9c0c47b1bb3a, do they change anything? different values passed to shaders, different formats of things?
```
I think doesn't change nothing for the games, its an internal remake but apparently the same facing out, in any case @lhog should know.

saurtron commented on doc/site/changelogs/changelog-2025-03.markdown:
```diff
@@ -171,6 +184,8 @@ The practical effect is that the Engine table is available in some LuaParser env
 * add `experience.experienceGrade` number modrule, same as calling Spring.SetExperienceGrade.
 * the `allowHoverUnitStrafing` modrule now defaults to `false`. Previously it defaulted to `false` for HAPFS and `true` for QTPFS.
 * bumpmapped water (aka `/water 4`) now has a different default texture.
+* add more data to desync dumps. The extra data will be dumped to files named `ClientGameState-<random number>-[<desync frame>-<desync frame>].txt`, i.e. same as before.
```
the data included is all the intra frame checksums, that means all the intermediate checksums from last frame with a good checksum, to the one that triggered the desync, that can be useful in better determining what exactly triggered the desync.

sprunk commented:
Updated.

Has some FIXMEs where I wasn't sure what to put or whether there was a change at all:
* unit shaders receive TRS transforms instead of matrices, breaking shaders. This will require gamedevs to update GL4 shaders that used to employ the engine provided transformation matrices SSBO. The new format is FIXME, and a more detailed documentation is found at FIXME.
* GLTF parser will insert a few empty pieces as root / sub-root. (FIXME whenever the commit message for ff592b0001fb62c6251ef23b6ed3d873da23414f says "GLTF" does it apply to GLB too?)
* the support for re-parenting is removed as well as the baked rotation angles. Use your 3d editor to change them instead. (FIXME was it removed from assimp lua too, or just not implemented for gltf? what tags precisely are affected?)
* added `AllowColorFonts` bool springsetting, defaults to false. Allows fonts to be colored instead of monochromatic. (FIXME does this effectively only apply to emoji? or can you also apply colors to "normal" text beyond what color codes let you do?)
* add `Spring.SetFeatureFireTime(featureID, fireTime)`, sets a feature on fire (similar to being damaged with a `firestarter` tagged weapon) with that much time to live, in seconds. FIXME does setting to 0 instakill or put out the fire?
* FIXME some info shader changes at commit be96560770b95cfe18a6c5ff3cbd9c0c47b1bb3a, do they change anything? different values passed to shaders, different formats of things?
* FIXME "infotex upgrade" at commit be96560770b95cfe18a6c5ff3cbd9c0c47b1bb3a, commit message says nothing. Did anything change for existing shaders, maybe different uniforms are passed or something?

And some mostly as reminders that the design looks possible questionable after putting it in words:
*  `BeamLaser` now obeys ellipsoid and/or cylinder target volumes correctly. (FIXME: not lightning cannon too?)
* added `animParams1` and `animParams2` to CEGs, similar to weapons. Texture 2 is only used for the "side" texture of `CBitmapMuzzleFlame`, all other particles (and "front" texture of muzzle flame) use texture 1. Not all CEG particles types support animations (FIXME: why, can't all textures get support? if not then list them, supported uses AddEffectsQuad<1> in implementation while unsupported is <0>). Numberless `animParams` used as a fallback.
* added `drawSideX` and `drawSizeY` bool tags to `CBitmapMuzzleFlame`, allows to skip some of the three sprites. (FIXME which is X and which is Y? why is there no Z, is this because you can just specify empty texture for Z but X and Y use the same tex? if so then why not allow separate tex for X and Y, then you could set one empty?)

badosu commented on doc/site/content/changelogs/changelog-2025-03.markdown:
```diff
@@ -2,27 +2,30 @@
 title = "Release 2025.03"
```
Maybe rename the filename and title?

badosu replied:
hm.. I'm concerned here, I'd rather have the filename accurate. Would you mind if I check and assure you:

- We have `2025-03` available so links don't break and point to `2025-04`?
- I fix any references to `2025-03` in site to `2025-04`?

If you do mind, then feel free to resolve this.

sprunk replied:
Sure, renamed.

I'd be happy if existing links to 03 keep working, but it's more about the principle and having some awareness/experience for handling this kind of rename in the future, rather than an expectation that such links actually exist in the wild and could see use.

badosu replied:
Yeah, I agree with you. See if the redirect I suggested [here](https://github.com/beyond-all-reason/RecoilEngine/pull/2365/files#r2267461820) makes sense.

See: https://gohugo.io/content-management/urls/#alias-examples

saurtron commented on doc/site/content/changelogs/_index.markdown:
```diff
@@ -40,11 +64,109 @@ See [the 2025.03 page]({{% ref "changelogs/changelog-2025-03" %}}) for the upcom
+
+### Text coloring
+
+- added `AllowColorFonts` bool springsetting, defaults to false. Allows fonts to be colored instead of monochromatic. (FIXME does this effectively only apply to emoji? or can you also apply colors to "normal" text beyond what color codes let you do?)
```
applies to any color font, ie, allows loading "color" font files with actual color instead of monochromatic (when `false` the font will still be loaded, but converted to grayscale).

I only know of emoji fonts doing this, but there might be others around, not sure.

saurtron commented on doc/site/content/changelogs/_index.markdown:
```diff
@@ -40,11 +64,109 @@ See [the 2025.03 page]({{% ref "changelogs/changelog-2025-03" %}}) for the upcom
+- add `ThreadPinPolicy` springsetting to control which CPUs to use. (0) = Off; (1) = System Default; (2) = Exclusive Performance Core; (3) = Share Performance Cores.
+- added new CEG impact filters, `shield` and `intercepted`.
+- `BeamLaser` and `LightningCannon` type weapons now pass their actual projectileID to callins rather than `-1`.
+- `BeamLaser` now obeys ellipsoid and/or cylinder target volumes correctly. (FIXME: not lightning cannon too?)
```
According to https://github.com/beyond-all-reason/RecoilEngine/issues/777 doesn't seem to be fixed for LightningCannon yet.

saurtron commented on doc/site/content/changelogs/_index.markdown:
```diff
@@ -5,21 +5,37 @@ title = "Running changelog"
+- `/group add N` no longer selects the entire group, just adds. `/groupN` (without space) unaffected. Feature support tag: `Engine.FeatureSupport.groupAddDoesntSelect`.
+- immobile units that `canKamikaze` no longer disregard the `blocking` unit def tag.
+- `BeamLaser` and `LightningCannon` type weapons now pass their actual projectileID to callins rather than `-1`.
+- `BeamLaser` now obeys ellipsoid and/or cylinder target volumes correctly. (FIXME: not lightning cannon too?)
```
not yet for LightningCannon according to https://github.com/beyond-all-reason/RecoilEngine/issues/777.

saurtron commented on doc/site/content/changelogs/_index.markdown:
```diff
@@ -40,11 +64,109 @@ See [the 2025.03 page]({{% ref "changelogs/changelog-2025-03" %}}) for the upcom
+- added `animParams1` and `animParams2` to CEGs, similar to weapons. Texture 2 is only used for the "side" texture of `CBitmapMuzzleFlame`, all other particles (and "front" texture of muzzle flame) use texture 1.
+  Not all CEG particles types support animations (FIXME: why, can't all textures get support? if not then list them, supported uses AddEffectsQuad<1> in implementation while unsupported is <0>).
+  Numberless `animParams` used as a fallback.
+- added `drawSideX` and `drawSizeY` bool tags to `CBitmapMuzzleFlame`, allows to skip some of the three sprites. (FIXME which is X and which is Y? why is there no Z, is this because you can just specify empty texture for Z but X and Y use the same tex? if so then why not allow separate tex for X and Y, then you could set one empty?)
```
I think `sideZ` is controlled through `frontTexture` existance, ie, as you say its "because you can just specify empty texture for Z but X and Y use the same tex".

more information at this comment https://github.com/beyond-all-reason/RecoilEngine/pull/2318#issuecomment-2955584466

regarding which is which, `side y` is the (non front) vertical, `side x` is the one parallel to ground, see https://github.com/beyond-all-reason/RecoilEngine/pull/2318#issuecomment-2955545452

badosu commented on doc/site/content/changelogs/changelog-2025-04.markdown:
```diff
@@ -1,28 +1,31 @@
 +++
-title = "Release 2025.03"
+title = "Release 2025.04"
```
```suggestion
title = "Release 2025.04"
aliases = ['/changelogs/changelog-2025-03']
```

sprunk commented on doc/site/content/changelogs/_index.markdown:
```diff
@@ -40,11 +64,109 @@ See [the 2025.03 page]({{% ref "changelogs/changelog-2025-03" %}}) for the upcom
+- add `Spring.CreateFeatureWreck(featureID, int? wreckLevel = 1, bool? emitSmoke = false) → featureID?`. Creates a wreck as if the unit died.
+- add `Spring.AddFeatureDamage` with the same signature as `Spring.AddUnitDamage`. Note that one of the args is paralyze time, but features don't take paralyze damage.
+- add a new `checkDeath` bool arg to `Spring.SetFeatureHealth(featureID, health, bool? checkDeath = false)`. It was discovered that the current behaviour is that features set to negative health don't die from HP removal and it takes an instance of damage to "poke" them into death. The default of `false` preserves this, set it to `true` to let features die immediately from HP removal.
+- add `Spring.SetFeatureFireTime(featureID, fireTime)`, sets a feature on fire (similar to being damaged with a `firestarter` tagged weapon) with that much time to live, in seconds. FIXME does setting to 0 instakill or put out the fire?
```
The "fire" name is consistent with other things that interact it ("firestarter", "burnable") so it's probably ok. If it's just a autodestroy timer and doesn't actually do anything related to fire then eventually we could provide a little lua gadget that reimplements it and remove it, for now it isn't too big of a burden though.

sprunk commented:
Pushed new version, updated remaining FIXMEs at https://github.com/beyond-all-reason/RecoilEngine/pull/2365#issuecomment-3152973451. Perhaps @lhog would know some of those now that he's back.
